soc/arc/snps_arc_hsdk/CMakeLists.txt

```diff
@@ -18,3 +18,5 @@ else()
 
   zephyr_ld_options(-Hlib=hs38_full)
 endif()
+
+set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "")
```

Comment:
Is the `CACHE INTERNAL ""` stuff really needed?  Maybe it's an optimization for incremental builds?  That's a little confusing for cmake amateurs like me, in a context where lots of cmake amateurs will be submitting code.

Isn't it possible to set attributes on a variable earlier, such that this can just reset the existing one?

Reply:
the problem is more related to scoping, as a value set in lower scope will not be reflected in (grand)-parent scopes.
You can set a value in parent, but not grand-parent scope (which we need).

The alternative would be to go through a property (requires the use of `set_property()` / `get_property()`) before use of a local var, but in this case I feel the use of an internal cache variable is cleaner, especially when using the var, as you can do:
```
if(CONFIG_HAVE_CUSTOM_LINKER_SCRIPT)
... 
elseif(DEFINED BOARD_LINKER_SCRIPT)
  ...
elseif(DEFINED SOC_LINKER_SCRIPT)
  ...
else()
  ...
endif()
```
instead of:
```
get_property(board_linker_script GLOBAL PROPERTY BOARD_LINKER_SCRIPT)
get_property(soc_linker_script GLOBAL PROPERTY SOC_LINKER_SCRIPT)

if(CONFIG_HAVE_CUSTOM_LINKER_SCRIPT)
... 
elseif(DEFINED board_linker_script )
  ...
elseif(DEFINED soc_linker_script )
  ...
else()
  ...
endif()

```